### PR TITLE
feat: Reset ul styles

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

In our app `<ul>` styles are not reset which is a pain as we want the markup but almost never the default styles.

## Changes

* Remove `ul` styles except when nested  in a `<p>` tag (where we assume it is prose)
* Fix a range of places where the `ul` styles were overridden as it is not necessary anymore
* Fixed `.list-` utils that never worked

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
:shipit:

File: frontend/src/scenes/authentication/PasswordReset.tsx [hunk 1]
```diff
@@ -55,16 +55,18 @@ function EmailUnavailable(): JSX.Element {
...
-                        <a href="https://posthog.com/docs/self-host/configure/email?utm_medium=in-product&utm_campaign=password-reset">
-                            Read the docs
-                        </a>{' '}
-                        on how to set this up.
-                    </li>
-                    <li>To reset the password manually, run the following command in your instance.</li>
-                </ul>
+                <p>
+                    <ul>
+                        <li>
```
Comment (mariusandra) [inline frontend/src/scenes/authentication/PasswordReset.tsx:60]: Can we put a `<ul>` in a `<p>`? I'm not sure that's semantically valid markup

> The short answer is that ol elements are not legally allowed inside p elements.
-- https://stackoverflow.com/questions/5681481/should-ol-ul-be-inside-p-or-outside

File: frontend/src/scenes/project/Settings/index.tsx [hunk 2]
```diff
@@ -195,16 +195,18 @@ export function ProjectSettings(): JSX.Element {
...
-                        "<strong>Email</strong> does not contain <strong>yourcompany.com</strong>" to exclude all events
-                        from your company's team members.
-                    </li>
-                    <li>
-                        "<strong>Host</strong> does not contain <strong>localhost</strong>" to exclude all events from
-                        local development environments.
-                    </li>
-                </ul>
+                <p>
+                    <ul>
```
Comment (mariusandra) [inline frontend/src/scenes/project/Settings/index.tsx:199]: Same question.

File: frontend/src/styles/global.scss [hunk 3]
```diff
@@ -25,6 +25,22 @@ body strong {
...
+}
+
+// ... unless in a <p> element
+p {
+    ul {
+        margin-bottom: 1rem;
+        list-style: initial;
+        padding-left: 1.5rem;
+    }
+}
```
Comment (mariusandra) [inline frontend/src/styles/global.scss:42]: I think [`ul.list-disc`](https://tailwindcss.com/docs/list-style-type) followed by `.space-y-2` or whatnot is a better solution than this _illegal_ solution? :)

**Review by mariusandra — CHANGES_REQUESTED** [review]
Well this got merged too soon. I was gonna ask to change some things... We shouldn't knowingly be writing invalid markup, even if it renders.

File: frontend/src/scenes/authentication/PasswordReset.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/authentication/PasswordReset.tsx:60]: ?? This is definitely valid and is a commonly used practice

**Pull request merged**